components/Header/index.tsx

```diff
@@ -32,6 +32,14 @@ const Header = () => {
   const mobileInputRef = useRef<HTMLInputElement>(null);
   const desktopInputRef = useRef<HTMLInputElement>(null);
 
+  useEffect(() => {
+    if (search.showSearch === true) {
```

Comment:
@ntorbinskiy Um, I really like that you are being, very explicit for comparison checks, but I guess in this case it is kind of clear, as showSearch is boolean type